### PR TITLE
Allow empty tables to be fitted using `fit_processed_data`

### LINKED ISSUE #1524: Allow empty tables to be fitted using fit_processed_data
### Problem Description

_As a user, it would be great of multi-table synthesizers were able to fit and sample even when child tables don't have statistical columns as that is a common database structure._

### Expected behavior

- `fit_processed_data` should not crash if the dataframe passed is empty.
- If the data is empty, it should:
    - set  `_fitted` to True
    - set `_fitted_date`
    - set `_fitted_sdv_version`
- If the data is not empty, it should only set those parameters if `_fit` runs successfully

### PR DESCRIPTION
**Pull request by fealho**
Resolve #1524.

### CODE REVIEW

File: tests/unit/multi_table/test_base.py [hunk 1]
```diff
@@ -718,6 +718,20 @@ def test_fit_processed_data(self):
...
+    def test_fit_processed_data_empty_table(self):
+        """Test the fit attributes are properly set when data is empty."""
+        # Setup
+        instance = Mock()
+        data = pd.DataFrame()
+
+        # Run
+        BaseMultiTableSynthesizer.fit_processed_data(instance, data)
+
+        # Assert
```
Comment (pvk-developer) [inline tests/unit/multi_table/test_base.py:730]: could we assert that `instance._fit` has not been called ?

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Review by amontanez24 — COMMENTED** [review]
Looks good! Same comment as @pvk-developer

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Pull request merged**